lib/util/http/index.ts

```diff
@@ -0,0 +1,111 @@
+import is from '@sindresorhus/is/dist';
+import URL from 'url';
+import got from '../got';
+
+interface OutgoingHttpHeaders {
+  [header: string]: number | string | string[] | undefined;
+}
+
+export interface HttpOptions {
+  auth?: string;
+  baseUrl?: string;
+  headers?: OutgoingHttpHeaders;
+  throwHttpErrors?: boolean;
+  useCache?: boolean;
+}
+
+export interface HttpPostOptions extends HttpOptions {
+  body: unknown;
+}
+
+interface InternalHttpOptions extends HttpOptions {
+  json?: boolean;
+  method?: 'get' | 'post';
+}
+
+export interface HttpResponse<T = string> {
+  body: T;
+  headers: any;
+}
+
+export class Http {
+  readonly hostType: string;
+
+  readonly options: HttpOptions;
+
+  constructor(hostType: string, options?: HttpOptions) {
+    this.hostType = hostType;
+    this.options = options;
+  }
```

Comment:
```suggestion

  constructor(private hostType: string, private options?: HttpOptions) { }
```
I think they should be private.